docs/sources/articles/dockerfile_best-practices.md

```diff
@@ -398,6 +398,14 @@ troubleshoot, and maintain.
 
 ### [`ONBUILD`](https://docs.docker.com/reference/builder/#onbuild)
 
+An `ONBUILD` command in a Dockerfile will not be executed when this
```

Comment:
Stating things in the negative and in the future can be hard to parse.  Instead, try this for easier reading.  You'll find the markdown in this gist:

https://gist.github.com/moxiegirl/b93d75a32a9c82cd03f3


-----
An `ONBUILD` command executes after the current Dockerfile build completes;
`ONBUILD` executes for any child image derived `FROM` the current image. Think
of the `ONBUILD` command as an instruction that parent Dockerfile gives to the child
Dockerfile.

A Docker build executes `ONBUILD` commands before any command in a child
Dockerfile.

`ONBUILD` is useful for images that are built `FROM` a given image. For example,
you would use `ONBUILD` for a language stack image that builds arbitrary user
software written in that language within the `Dockerfile`, as you can see in
[Ruby’s `ONBUILD`
variants](https://github.com/docker-library/ruby/blob/master/2.1/onbuild/
Dockerfile).

Tag images built from `ONBUILD` separately from the parent, for example:
`ruby:1.9-onbuild` or `ruby:2.0-onbuild`.

Be careful when putting `ADD` or `COPY` in `ONBUILD`. The “onbuild” image can
fail catastrophically if the new build's context is missing a resource being
added or copied. Adding a separate tag, as recommended above, may help mitigate
this by allowing the `Dockerfile` author to make a choice.